macro.js

```diff
@@ -20,10 +20,36 @@ function prevalMacros({ references, state, babel }) {
 }
 
 function asTag(quasiPath, { file: { opts: { filename } } }, babel) {
-  const string = quasiPath.parentPath.get("quasi").evaluate().value;
   const { types: t } = babel;
+  
+  const originalQuasis = quasiPath.get('quasis').map(quasi => quasi.node);
+  const strings = { raw: originalQuasis.map(quasi => quasi.value.raw) };
+  const expressions = quasiPath.get('expressions').map(expression => expression.node);
+  
+  // If no expressions, use string literal
+  // otherwise, use template literal
+  let replacement
+  if (!expressions.length) {
+    replacement = t.stringLiteral(dedent(strings));
+  } else {
+    const EXPRESSION = '---EXPR---';
 
-  quasiPath.parentPath.replaceWith(t.stringLiteral(dedent(string)));
+    const originalQuasis = quasiPath.get('quasis').map(quasi => quasi.node);
+    const expressions = quasiPath.get('expressions').map(expression => expression.node);
+    
+    const strings = { raw: originalQuasis.map(quasi => quasi.value.raw) };
+    const placeholders = expressions.map(() => EXPRESSION);
+    const result = dedent.apply(dedent, [strings].concat(placeholders));
+
+    const {types: t } = babel;
+    const quasis = result.split(EXPRESSION).map((value, index) => {
+      return t.templateElement({ raw: value, cooked: value }, originalQuasis[index].tail);
+    });
+
+    replacement = t.templateLiteral(quasis, expressions);
+  }
+
+  quasiPath.parentPath.replaceWith(replacement);
```

Comment:
> Add fallback to macro for compiling into template literal when evaluate fails (e.g. external variables)

Sorry, why do we want this? I'm familiar enough with Babel macros, could you post a deeper explanation than just this sentence + tests please?